src/classes/RD2_Schedule_SVC.cls

```diff
@@ -0,0 +1,185 @@
+        }
+
+        for (RecurringDonationSchedule__c rds : schedulesToInsert) {
+            rds.StartDate__c = calculateStartDate(rds, rd);
+        }
+
+        return this;
+    }
+
+    // TODO: Update schedule to reflect recurringDonation - W-036113 (Placeholder)
+    public RD2_Schedule_SVC updateSchedule(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdScheds) {
+        return this;
+    }
+
+    // TODO: Will return next Donation date - W-036251 (Placeholder)
+    public Date getNextDonationDate(RecurringDonationSchedule__c rdSched) {
+        return null;
+    }
+
+    // TODO: Will return an inner class (Placeholder - RDScheduleDates)
+    public List<RDScheduleDates> visualizeSchedule(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> rdScheds) {
+        return new List<RDScheduleDates>();
+    }
+
+    // TODO: Placeholder internal class used for visualization)
+    public class RDScheduleDates {
+        public npe03__Recurring_Donation__c recurringDonation;
+        public List<RecurringDonationSchedule__c> rdSchedules;
+        public Date donationDate;
+    }
+
+    /***
+    * @description Calculates start date for a recurring donation schedule
+    * @param rdSchedule Recurring Donation schedule
+    * @param rd Recurring Donation schedule
+    * @return Date
+    */
+    private Date calculateStartDate(RecurringDonationSchedule__c rdSchedule, npe03__Recurring_Donation__c rd) {
+        Date retDate;
+
+        Date minSD = rd.StartDate__c > rd.npe03__Date_Established__c ?
+                     rd.StartDate__c :
+                     rd.npe03__Date_Established__c;
+
+        retDate = minSD > Date.today() ?
+                  minSD :
+                  Date.today();
+
+        if (rdSchedule.InstallmentPeriod__c == MONTHLY) {
+            Integer lastDay = Date.daysInMonth(retDate.year(), retDate.month());
+            retDate = Date.newInstance(retDate.year(), retDate.month(), rdSchedule.DayOfMonth__c == LAST_DAY_OF_MONTH ?
+                                                                        lastDay :
+                                                                        Integer.valueOf(rdSchedule.DayOfMonth__c));
+            retDate = retDate >= Date.today() ?
+                      retDate :
+                      retDate.addMonths(1);
+        }
+
+        return retDate;
+    }
+
+    /***
+    * @description Determines whether '1st and 15th' is the schedule selected.
+    * @param rd Recurring Donation record
+    * @return Boolean
+    */
+    private Boolean isFirstAndFifteenth(npe03__Recurring_Donation__c rd) {
+        return rd.Day_of_Month__c == FIRST_AND_FIFTEENTH && rd.npe03__Installment_Period__c == MONTHLY;
+    }
+
+    /***
+    * @description Gets schedule day of month from recurring donation day of month
+    * @param rd Recurring Donation
+    * @return String
+    */
+    private String getDayOfMonth(npe03__Recurring_Donation__c rd) {
```

Comment:
Based on usage from above, I don't think this method is necessary. The DayOfMonth value on the Schedule record should always be either an exact copy of the RD.Day_Of_Month__c field value or null (for Weekly and Yearly schedules).